### ISSUE TITLE
Service Catalog Upgrade fails if executed second time

### ISSUE THREAD

**Issue by mszostok** [kyma-project/kyma#5869]
**Description**

Currently, the Service Catalog upgrade action can be executed only once.  When it is executed the second time then such err is returned: 

```
service-catalog:
  Helm upgrade error: rpc error: code = Unknown desc = persistentvolumeclaims "service-catalog-catalog-migration-storage" already exists
```

The problem is also that in first iteration the Service Catalog resources cannot be restored because of temporary CRD unavailability in main api server.

**Comment by mszostok** [comment]
As a first step we will and waiter for ready CRDs definition and also retries on restore operation.

If such workaround will fail then we will try to figure out the way for handling the multi upgrade execution

**Comment by mszostok** [comment]
This issue was solved by updating the Service Catalog to v0.3.0-beta.1 where discussed fixes were merged, check this PR for more info: https://github.com/kyma-project/kyma/pull/5879
